### PR TITLE
[ParseGen] Correctly set `MACOSX_DEPLOYMENT_TARGET`

### PR DESCRIPTION
**Pull request by jeremiahpslewis**

### CODE REVIEW

File: P/ParseGen/bundled/sw_vers [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
```
Comment (giordano) [inline P/ParseGen/bundled/sw_vers:1]: Isn't it easier to just set MACOSX_DEPLOYMENT_TARGET?

File: P/ParseGen/build_tarballs.jl [hunk 2]
```diff
@@ -9,19 +9,19 @@ version = v"2.0.0"
...
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/parsegen-cpp*/
 
 install_license LICENSE
-
 mkdir build && cd build
 
 if [[ "${target}" == x86_64-apple-darwin* ]]; then
     export CXXFLAGS="-mmacosx-version-min=10.15"
```
Comment (giordano) [inline P/ParseGen/build_tarballs.jl:23]: You likely don't need this line. And the entire aarch64-apple-darwin branch is now useless

Comment (giordano) [reply P/ParseGen/build_tarballs.jl:23]: This isn't resolved. I'm still convinced this line isn't needed.

**Review by giordano — CHANGES_REQUESTED** [review]
I'm not sure what's exactly the point of this pull request which doesn't change anything in the resulting tarballs nor the generated JLL package.

If you just want to make it work with newer cmake, which is fine, please use `[skip ci]` because I don't want this to trigger a rebuild of anything after merge.

File: P/ParseGen/build_tarballs.jl [hunk 3]
```diff
@@ -9,28 +9,25 @@ version = v"2.0.0"
...
```
Comment (jeremiahpslewis) [inline P/ParseGen/build_tarballs.jl:23]:
```suggestion
```

File: P/ParseGen/build_tarballs.jl (hunk 2 above)
Comment (jeremiahpslewis) [reply P/ParseGen/build_tarballs.jl]: Thanks, didn't see the line you were referring to the first time.

**Comment by jeremiahpslewis** [comment]
> I'm not sure what's exactly the point of this pull request which doesn't change anything in the resulting tarballs nor the generated JLL package.
> 
> If you just want to make it work with newer cmake, which is fine, please use `[skip ci]` because I don't want this to trigger a rebuild of anything after merge.

I was looking for a clean working example of a Mac x86 10.14 build to debug a regression which appeared with BinaryBuilderBase 1.6.0 so I could figure out another binary.

**Comment by jeremiahpslewis** [comment]
Added [skip ci]

**Pull request merged**